test/hotspot/jtreg/runtime/modules/IgnoreModulePropertiesTest.java

```diff
@@ -58,21 +58,27 @@ public static void testProperty(String prop, String value) throws Exception {
     // For options of the form "option=value", check that an exception gets thrown for
     // the illegal value and then check that its corresponding property is handled
     // correctly.
-    public static void testOption(String option, String value,
+    public static void testOption(boolean positive,
+                                  String option, String value,
                                   String prop, String result) throws Exception {
         ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
             option + "=" + value, "-version");
         OutputAnalyzer output = new OutputAnalyzer(pb.start());
+        if (positive) {
+            output.shouldHaveExitValue(0);
+        } else {
+            output.shouldNotHaveExitValue(0);
+        }
         output.shouldContain(result);
         testProperty(prop, value);
     }
 
     public static void main(String[] args) throws Exception {
-        testOption("--add-modules", "java.sqlx", "jdk.module.addmods.0", "java.lang.module.FindException");
-        testOption("--limit-modules", "java.sqlx", "jdk.module.limitmods", "java.lang.module.FindException");
-        testOption("--add-reads", "xyzz=yyzd", "jdk.module.addreads.0", "WARNING: Unknown module: xyzz");
-        testOption("--add-exports", "java.base/xyzz=yyzd", "jdk.module.addexports.0",
+        testOption(false, "--add-modules", "java.sqlx", "jdk.module.addmods.0", "java.lang.module.FindException");
```

Comment:
It would be a litle more obvious if you did:

`final boolean shouldSucceed = true;`

and then do either:

`testOption(!shouldSucceed, ...);`

or

`testOption(shouldSucceed, ...);`

Reply:
not sure if `testOption(!shouldSucceed,...)` looks/reads nice, I can do `testOption(/*shouldSucceed=*/true, ...);` instead.

Reply:
Okay